src/planscape/planning/views.py

```diff
@@ -366,36 +370,100 @@ def get_scenario_download_by_id(request: HttpRequest) -> HttpResponse:
 
     TODO: maybe generate a unique key and store that for each output dir name when we create it?
     """
-    try:
-        # Ensure that the user is logged in.
-        user = _get_user(request)
-        if user is None:
-            return HttpResponseBadRequest(
-                "Unauthorized. User is not logged in.", status=401
-            )
+    # Ensure that the user is logged in.
+    user = _get_user(request)
+    if user is None:
+        return HttpResponse(
+            "Unauthorized. User is not logged in.",
+            status=401,
+        )
 
-        scenario = Scenario.objects.select_related("planning_area__user").get(
-            id=request.GET["id"]
+    scenario = (
+        Scenario.objects.select_related("planning_area__user")
+        .filter(id=request.GET["id"])
+        .first()
+    )
+
+    if not scenario:
+        return HttpResponse(
+            "Scenario matching query does not exist.",
+            status=404,
+        )
+
+    # Ensure that current user is associated with this scenario
+    if scenario.planning_area.user.pk != user.pk:
+        return HttpResponse(
+            "Scenario matching query does not exist.",
+            status=404,
         )
-        # Ensure that current user is associated with this scenario
-        if scenario.planning_area.user.pk != user.pk:
-            raise ValueError("Scenario matching query does not exist.")
 
-        scenario_result = ScenarioResult.objects.get(scenario__id=scenario.pk)
-        if scenario_result.status != ScenarioResultStatus.SUCCESS:
-            raise ValueError("Scenario was not successful, data cannot be downloaded.")
+    scenario_result = ScenarioResult.objects.get(scenario__id=scenario.pk)
+    if scenario_result.status != ScenarioResultStatus.SUCCESS:
+        return HttpResponse(
+            "Scenario was not successful, data cannot be downloaded.",
+            status=409,
```

Comment:
I thought about returning 409 here, too, but I think this is only meant to refer to a conflict with an already existing resource that [ _does_ exist](https://www.abstractapi.com/http-status-codes/409), but can't be used because another process has modified it.

I was also considering a 422 or maybe even the very rare [424](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/424), but maybe just a regular old 400 or 404 works here?

Reply:
I guess my gut is saying 424 here, but I suppose a lot of these are valid.